jupyter_kernel_mgmt/subproc/manager.py

```diff
@@ -113,5 +138,13 @@ def __init__(self, popen, files_to_cleanup=None, win_interrupt_evt=None):
 
     async def is_alive(self):
         """Is the kernel process still running?"""
-        return not self._exit_future.done()
+        if self.async_subprocess:
+            return not self._exit_future.done()
+        else:
+            is_alive = False  # assume the kernel is dead
+            if self.kernel is not None:
+                if self.kernel.poll() is None:
+                    is_alive = True
```

Comment:
As we're waiting for Min's patch anyway, I think there's a more concise way to do this:

```python
return self.kernel and (self.kernel.poll() is None)
```

(The parentheses are probably unnecessary, but I find it easier to overuse them than to learn operator precedence ;-)

Reply:
Agreed.